hooks/useConnectWallet.tsx

```diff
@@ -1,22 +1,30 @@
-import trackLoginPoints from "@/lib/stack/trackLoginPoints";
-import { useEffect } from "react";
-import { useAccount, useConnect } from "wagmi";
+import { usePrivy } from '@privy-io/react-auth';
+import { Address } from "viem";
 
-const useConnectWallet = () => {
-  const { connectors, connectAsync } = useConnect();
-  const connector = connectors[0];
-  const { address } = useAccount();
+interface ConnectWalletReturn {
+  connectWallet: () => void;
+  address: Address | undefined;
+  authenticated: boolean;
+  ready: boolean;
+  login: () => void;
+}
 
-  useEffect(() => {
-    if (!address) return;
-    trackLoginPoints(address);
-  }, [address]);
+const useConnectWallet = (): ConnectWalletReturn => {
+  const { login, ready, authenticated, user } = usePrivy();
+  const address = user?.wallet?.address;
 
-  const connectWallet = () => connectAsync({ connector });
+  const connectWallet = () => {
+    if (!authenticated && ready) {
+      return login();
+    }
+  };
 
   return {
     connectWallet,
-    address,
+    address: address as Address | undefined,
+    authenticated,
+    ready,
+    login
```

Comment:
Why export these from `useConnectWallet`?
- I could just import directly from `usePrivy` right?
- What value is this export adding to our codebase?